app/utils/geostore.py

```diff
@@ -1,25 +1,40 @@
 from uuid import UUID
 
+from async_lru import alru_cache
 from fastapi import HTTPException
 
+from app.crud.geostore import get_geostore_from_anywhere
 from app.errors import BadResponseError, InvalidResponseError
 from app.models.enum.geostore import GeostoreOrigin
+from app.models.pydantic.geostore import Geometry, GeostoreHydrated
 from app.utils import rw_api
 
 
+@alru_cache(maxsize=128)
+async def _get_gfw_geostore_geometry(geostore_id: UUID) -> Geometry:
+    """Get GFW Geostore geometry."""
+
+    geo: GeostoreHydrated = await get_geostore_from_anywhere(geostore_id)
+
+    try:
+        geometry = geo.gfw_geojson.features[0].geometry
+    except KeyError:
```

Comment:
Also need to catch `RecordNotFoundError` error and raise a `BadResponseError`, if geostore_id does not return a match